tst/unit/test_unit_integrators.cpp

```diff
@@ -0,0 +1,251 @@
+}
+
+template <typename Integrator, typename Stepper>
+void Integrate(const Integrator &integrator, const Stepper &step, const Real tf, Real dt,
+               State_t &u0) {
+  assert(tf > 0);
+  assert(dt > 0);
+  assert(dt < tf);
+  // stupid game to align to EXACTLY tf
+  int NT = std::ceil(tf / dt);
+  Real t = 0;
+  for (int i = 0; i < NT; ++i) {
+    step(integrator, dt, u0);
+    t += dt;
+  }
+  if ((t < tf) && (std::abs(tf - t) > 1e-12)) {
+    dt = tf - t;
+    step(integrator, dt, u0);
+    t += dt;
+  }
+  // printf("%.14e %.14e\n", t, tf);
+}
+
+TEST_CASE("Low storage integrator", "[StagedIntegrator]") {
+  GIVEN("A state with an initial condition") {
+    Real t0 = 0, tf = 1.15; // delibarately not a nice fraction of a period
+    State_t ufinal;
+    GetTrueSolution(tf, ufinal);
+    WHEN("We integrate with LowStorage rk1") {
+      constexpr Real dt = 1e-5;
+      auto integrator = MakeIntegrator<LowStorageIntegrator>("rk1");
+      State_t u;
+      GetInitialData(u);
+      Integrate(integrator, Step2SStar, tf, dt, u);
+      THEN("The final state doesn't differ too much from the true solution") {
+        REQUIRE(std::abs(u[0] - ufinal[0]) <= 1e-2);
+        REQUIRE(std::abs(u[1] - ufinal[1]) <= 1e-2);
+      }
+    }
+    WHEN("We integrate with LowStorage rk2") {
+      constexpr Real dt = 1e-3;
+      auto integrator = MakeIntegrator<LowStorageIntegrator>("rk2");
+      State_t u;
+      GetInitialData(u);
+      Integrate(integrator, Step2SStar, tf, dt, u);
+      THEN("The final state doesn't differ too much from the true solution") {
+        REQUIRE(std::abs(u[0] - ufinal[0]) <= 1e-2);
+        REQUIRE(std::abs(u[1] - ufinal[1]) <= 1e-2);
+      }
+    }
+    WHEN("We integrate with LowStorage vl2") {
+      constexpr Real dt = 1e-3;
+      auto integrator = MakeIntegrator<LowStorageIntegrator>("vl2");
+      State_t u;
+      GetInitialData(u);
+      Integrate(integrator, Step2SStar, tf, dt, u);
+      THEN("The final state doesn't differ too much from the true solution") {
+        REQUIRE(std::abs(u[0] - ufinal[0]) <= 1e-2);
+        REQUIRE(std::abs(u[1] - ufinal[1]) <= 1e-2);
+      }
+    }
+    WHEN("We integrate with LowStorage rk3") {
+      constexpr Real dt = 1e-2;
+      auto integrator = MakeIntegrator<LowStorageIntegrator>("rk3");
+      State_t u;
+      GetInitialData(u);
+      Integrate(integrator, Step2SStar, tf, dt, u);
+      THEN("The final state doesn't differ too much from the true solution") {
+        REQUIRE(std::abs(u[0] - ufinal[0]) <= 1e-2);
+        REQUIRE(std::abs(u[1] - ufinal[1]) <= 1e-2);
+      }
+    }
+    WHEN("We integrate with LowStorage rk4") {
+      constexpr Real dt = 1e-2;
+      auto integrator = MakeIntegrator<LowStorageIntegrator>("rk4");
+      State_t u;
+      GetInitialData(u);
+      Integrate(integrator, Step2SStar, tf, dt, u);
+      THEN("The final state doesn't differ too much from the true solution") {
+        REQUIRE(std::abs(u[0] - ufinal[0]) <= 1e-2);
```

Comment:
thresholds differ for the Butcher integrators, however, low storage RK4 has the same threshold as lower storage RK1

Reply:
Yes---that's because I chose a very large timestep. timestep for RK1 is 1e-5. Timestep for rk4 is 1e-2. I'll add a comment.